src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -699,17 +699,41 @@ static ucs_status_t uct_rc_mlx5_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
-                                   const uct_device_addr_t *dev_addr,
-                                   const uct_iface_addr_t *iface_addr)
+int uct_rc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                      const uct_iface_is_reachable_params_t *params)
 {
-    uint8_t my_type = uct_rc_mlx5_iface_get_address_type(tl_iface);
+    const uct_iface_addr_t *iface_addr = params->iface_addr;
+    char *info_string                  = params->info_string;
+    size_t info_string_length          = params->info_string_length;
+    uct_ib_iface_t *iface              = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    uint8_t my_type                    = uct_rc_mlx5_iface_get_address_type(tl_iface);
 
     if ((iface_addr != NULL) && (my_type != *(uint8_t*)iface_addr)) {
+        snprintf(info_string, info_string_length,
+                 "IB iface %p: is unreachable due to address type mismatch, "
+                 "local=%d, remote=%d", iface, my_type, *(uint8_t*)iface_addr);
         return 0;
     }
 
-    return uct_ib_iface_is_reachable(tl_iface, dev_addr, iface_addr);
+    return uct_ib_iface_is_reachable_v2(tl_iface, params);
+}
+
+int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
+                                   const uct_device_addr_t *dev_addr,
+                                   const uct_iface_addr_t *iface_addr)
+{
+    const uct_iface_is_reachable_params_t params = {
+        .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
+        .device_addr        = dev_addr,
+        .iface_addr         = iface_addr,
+        .info_string        = NULL,
+        .info_string_length = 0
+    };
+    
+    return uct_rc_mlx5_iface_is_reachable_v2(tl_iface, &params);
```

Comment:
I think we can avoid this code duplication adding helper function with
```
    const uct_iface_is_reachable_params_t params = {
        .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
                              UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING |
                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
        .device_addr        = dev_addr,
        .iface_addr         = iface_addr,
        .info_string        = NULL,
        .info_string_length = 0
    };

    return uct_iface_is_reachable_v2(tl_iface, &params);
```

Reply:
The 2nd parameter of uct_iface_is_reachable_v2() function is a const type. `params` should be declare as non-const if we use a common function to initialize it. So, we need to cast it as const when we call uct_iface_is_reachable_v2, just like:
```
uct_iface_is_reachable_params_t params;

uct_iface_reachable_params_init(&params); 

return uct_iface_is_reachable_v2(tl_iface, (const uct_iface_is_reachable_params_t*)&params);
```

That's OK?
Otherwise the CI will be failed

Reply:
no, I meant common v1 to v2 wrapper

Reply:
Done. Please review. Thanks